test/e2e/validators/numericValidator.spec.js

```diff
@@ -111,6 +111,49 @@ describe('numericValidator', () => {
     }, 100);
   });
 
+  it('should validate large-number scientific notation', (done) => {
+    var onAfterValidate = jasmine.createSpy('onAfterValidate');
+
+    handsontable({
+      data: arrayOfObjects(),
+      columns: [
+        {data: 'id', type: 'numeric'},
+        {data: 'name'},
+        {data: 'lastName'}
+      ],
+      afterValidate: onAfterValidate
+    });
+
+    setDataAtCell(2, 0, '1e+23');
+
+    setTimeout(() => {
+      expect(onAfterValidate).toHaveBeenCalledWith(true, 1e+23, 2, 'id', undefined, undefined);
+      done();
+    }, 100);
+  });
+
+
+  it('should validate small-number scientific notation', (done) => {
+    var onAfterValidate = jasmine.createSpy('onAfterValidate');
+
+    handsontable({
+      data: arrayOfObjects(),
+      columns: [
+        {data: 'id', type: 'numeric'},
+        {data: 'name'},
+        {data: 'lastName'}
+      ],
+      afterValidate: onAfterValidate
+    });
+
+    setDataAtCell(2, 0, '1e-23');
+
+    setTimeout(() => {
+      expect(onAfterValidate).toHaveBeenCalledWith(true, 1e-23, 2, 'id', undefined, undefined);
+      done();
+    }, 100);
+  });
+
```

Comment:
I'm not concerned with that case, as long as 1.2e+23 is validated as a number. I would suggest you not include tests of ECMAScript-based functionality along with your application.

Reply:
I didn't mean adding a test of ECMAScript-based functionality :)

I thought, that NumericEditor under the hood is a `<textarea>` element which always contains value as a number. Inserted `120000000000000000000000` in NumericEditor should be saved as `1.2e+23` in datasource.
What do you think about my idea?

Reply:
Let me clarify:

1.2e+23 and 120000000000000000000000 are exactly the same thing as far as Javascript is concerned. So if you write the test that "if you enter 120000000000000000000000 you get 1.2e+23," it's the same thing as asking "if you enter 2, do you get 2." 

If you want to ensure that 120000000000000000000000 is always rendered as 1.2e+23, that's where you're testing whether the Javascript environment you're working in is following ECMAScript guidelines; again probably something you don't want to be doing. 

Whereas the rest of this code is asking: if I enter 120000000000000000000000, does it validate as a number correctly.

So we can add this test if you want, but it will never fail.

Reply:
I understand what do you mean. I thought about this effect: http://jsfiddle.net/dzw4jq81/
The first time, when you open an editor, the value is still `120000000000000000000000`.
Press <kbd>ENTER</kbd> twice.
End-users will see something different than they've inserted.

But I think it's topic to discuss in our @handsontable/core team.
Please resolve conflicts and I will merge it to develop.